File: backend/src/main/java/org/luxons/sevenwonders/actions/CreateGameAction.java
package org.luxons.sevenwonders.actions;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import org.hildan.livedoc.core.annotations.types.ApiType;

/**
 * The action to create a game.
 */
@ApiType(group = "Actions")
public class CreateGameAction {

    @NotNull
    @Size(min = 2, max = 30)
    private String gameName;

    /**
     * @return The name of the game to create
     */
    public String getGameName() {
        return gameName;
    }

    public void setGameName(String gameName) {
        this.gameName = gameName;
    }
}

File: sw-server/src/main/kotlin/org/luxons/sevenwonders/actions/CreateGameAction.kt
package org.luxons.sevenwonders.actions

import org.hildan.livedoc.core.annotations.types.ApiType
import org.hildan.livedoc.core.annotations.types.ApiTypeProperty
import org.luxons.sevenwonders.doc.Documentation
import javax.validation.constraints.Size

/**
 * The action to create a game.
 */
@ApiType(group = Documentation.GROUP_ACTIONS)
class CreateGameAction(
    /**
     * The name of the game to create.
     */
    @Size(min = 2, max = 30)
    @ApiTypeProperty(required = true)
    val gameName: String
)
